README.mkdn

```diff
@@ -180,6 +180,140 @@ more tasks in the context of a particular image:
 $ cargo xtask clippy app/gimletlet/app.toml ping pong
 ```
 
+## Integrating with `rust-analyzer`
+The Hubris build system will not work with `rust-analyzer` out of the box.
+
+However, `cargo xtask lsp` is here to help: it takes as its argument a Rust
+file, and returns JSON-encoded configuration for how to set up `rust-analyzer`.
+
+To use this data, some editor configuration is required!
+
+(we haven't made plugins yet, but it would certainly be possible)
+
+Using Neovim and [`rust-tools`](https://github.com/simrat39/rust-tools.nvim),
+here's an example configuration:
+
+```lua
+-- monkeypatch rust-tools to correctly detect our custom rust-analyzer
+require'rust-tools.utils.utils'.is_ra_server = function (client)
+  local name = client.name
+  local target = "rust_analyzer"
+  return string.sub(client.name, 1, string.len(target)) == target
+    or client.name == "rust_analyzer-standalone"
+end
+
+-- Configure LSP through rust-tools.nvim plugin, with lots of bonus
+-- content for Hubris compatibility
+local cache = {}
+local clients = {}
+require'rust-tools'.setup{
+  tools = { -- rust-tools options
+    autoSetHints = true,
+    inlay_hints = {
+      show_parameter_hints = false,
+      parameter_hints_prefix = "",
+      other_hints_prefix = "",
+      -- do other configuration here as desired
+    },
+  },
+
+  server = {
+    on_new_config = function(new_config, new_root_dir)
+      local bufnr = vim.api.nvim_get_current_buf()
+      local bufname = vim.api.nvim_buf_get_name(bufnr)
+      local dir = new_config.root_dir()
+      if string.find(dir, "hubris") then
+        -- Run `xtask lsp` for the target file, which gives us a JSON
+        -- dictionary with bonus configuration.
+        local prev_cwd = vim.fn.getcwd()
+        vim.cmd("cd " .. dir)
+        local cmd = dir .. "/target/debug/xtask lsp "
```

Comment:
That was intentional, because running through `cargo xtask` takes ~580 ms; running the binary directly takes ~200 ms, which is very noticeable when you change a file.

I added a comment in 98474fc361b61930243b183b4389c4f0f6fc1a11 and also tweaked the suggested config to print an obvious warning.